src/sync/sync_manager.cpp

```diff
@@ -337,22 +337,20 @@ std::shared_ptr<SyncUser> SyncManager::get_user(const std::string& user_id,
         if (!m_metadata_manager)
             m_current_user = new_user;
         return new_user;
-    } else {
+    }
+    else { // LoggedOut || Removed => LoggedIn
         auto user = *it;
```

Comment:
I think the issue here is that the `find_if` is returning a user and then it is hitting this `else` case, where what should really happen is that no user gets returned at all. Maybe the solution lies within the search conditions:

```
auto it = std::find_if(m_users.begin(),
                           m_users.end(),
                           [user_id, provider_type](const auto& user) {
        return user->identity() == user_id && user->provider_type() == provider_type && user->state() != SyncUser::State::Removed;
    });
```

Reply:
Thanks for the suggestion! I looked into this and there are a few subtle differences. 
In your suggestion, a new user object instance will be created after the second log in, although the underlying sync metadata does revive it's representation in either case (removed -> logged_in). In the changes I proposed, the original user object remains valid after the second log in and changes it's state from Removed to LoggedIn.
So I guess the question is, what is appropriate behaviour here from a user or SDK perspective?

Reply:
From a lifecycle perspective, reviving a removed user seems incorrect. This user is marked to be discarded. A new user being created after login is fine from my point of view, as it just invokes the user lifecycle from the beginning.

Reply:
Thanks for confirming, I agree this makes the most sense. I've updated the code to reflect this.